Index: mythplugins/mythmusic/mythmusic/playbackbox.cpp
===================================================================
--- mythplugins/mythmusic/mythmusic/playbackbox.cpp	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/playbackbox.cpp	(working copy)
@@ -365,6 +365,8 @@
             changeSpeed(true);
         else if (action == "MUTE")
             toggleMute();
+        else if (action == "TOGGLEUPMIX")
+            toggleUpmix();
         else if (action == "MENU" && visualizer_status != 2)
         {
             menufilters = false;
@@ -1202,6 +1204,13 @@
     }
 }
 
+void PlaybackBoxMusic::toggleUpmix()
+{
+    if (gPlayer->getOutput())
+        gPlayer->getOutput()->ToggleUpmix();
+}
+    
+
 void PlaybackBoxMusic::showProgressBar()
 {
     if (progress_bar && visualizer_status != 2)
Index: mythplugins/mythmusic/mythmusic/globalsettings.cpp
===================================================================
--- mythplugins/mythmusic/mythmusic/globalsettings.cpp	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/globalsettings.cpp	(working copy)
@@ -67,6 +67,17 @@
     return gc;
 };
 
+static HostCheckBox *MusicUpmixer()
+{
+    HostCheckBox *gc = new HostCheckBox("MusicDefaultUpmix");
+    gc->setLabel(QObject::tr("Upconvert stereo to 5.1 surround"));
+    gc->setValue(false);
+    gc->setHelpText(QObject::tr("MythTV can upconvert stereo tracks to 5.1 audio. "
+                                "Set this option to enable it by default. "
+                                "You can enable or disable the upconversion during playback at anytime."));
+    return gc;
+};
+
 static HostLineEdit *CDDevice()
 {
     HostLineEdit *gc = new HostLineEdit("CDDevice");
@@ -545,6 +556,7 @@
     general->setLabel(QObject::tr("General Settings (1)"));
     general->addChild(SetMusicDirectory());
     general->addChild(MusicAudioDevice());
+    general->addChild(MusicUpmixer());
     general->addChild(CDDevice());
     general->addChild(AutoLookupCD());
     general->addChild(AutoPlayCD());
Index: mythplugins/mythmusic/mythmusic/avfdecoder.cpp
===================================================================
--- mythplugins/mythmusic/mythmusic/avfdecoder.cpp	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/avfdecoder.cpp	(working copy)
@@ -256,8 +256,8 @@
     if (output())
     {
         const AudioSettings settings(
-            16 /*bits*/, m_audioDec->channels, m_audioDec->sample_rate,
-            false /* AC3/DTS pass through */);
+            16 /*bits*/, m_audioDec->channels, m_audioDec->codec_id, 
+            m_audioDec->sample_rate, false /* AC3/DTS pass through */);
         output()->Reconfigure(settings);
         output()->SetSourceBitrate(m_audioDec->bit_rate);
     }
Index: mythplugins/mythmusic/mythmusic/main.cpp
===================================================================
--- mythplugins/mythmusic/mythmusic/main.cpp	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/main.cpp	(working copy)
@@ -389,6 +389,7 @@
     REG_KEY("Music", "VOLUMEDOWN", "Volume down",       "[,{,F10,Volume Down");
     REG_KEY("Music", "VOLUMEUP",   "Volume up",         "],},F11,Volume Up");
     REG_KEY("Music", "MUTE",       "Mute",              "|,\\,F9,Volume Mute");
+    REG_KEY("Music", "TOGGLEUPMIX","Toggle upmixer",             "Ctrl+U");
     REG_KEY("Music", "CYCLEVIS",   "Cycle visualizer mode",      "6");
     REG_KEY("Music", "BLANKSCR",   "Blank screen",               "5");
     REG_KEY("Music", "THMBUP",     "Increase rating",            "9");
Index: mythplugins/mythmusic/mythmusic/musicplayer.cpp
===================================================================
--- mythplugins/mythmusic/mythmusic/musicplayer.cpp	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/musicplayer.cpp	(working copy)
@@ -348,16 +348,19 @@
 
 void MusicPlayer::openOutputDevice(void)
 {
-    QString adevice;
+    QString adevice, pdevice;
 
     if (gContext->GetSetting("MusicAudioDevice") == "default")
         adevice = gContext->GetSetting("AudioOutputDevice");
     else
         adevice = gContext->GetSetting("MusicAudioDevice");
 
+    pdevice = gContext->GetSetting("PassThruOutputDevice");
+
     // TODO: Error checking that device is opened correctly!
-    m_output = AudioOutput::OpenAudio(adevice, "default", 16, 2, 44100,
-                                    AUDIOOUTPUT_MUSIC, true, false);
+    m_output = AudioOutput::OpenAudio(adevice, pdevice, 16, 2, 0, 44100,
+                                      AUDIOOUTPUT_MUSIC, true, false,
+                                      gContext->GetNumSetting("MusicDefaultUpmix", 0) + 1);
     m_output->setBufferSize(256 * 1024);
     m_output->SetBlocking(false);
 
Index: mythplugins/mythmusic/mythmusic/playbackbox.h
===================================================================
--- mythplugins/mythmusic/mythmusic/playbackbox.h	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/playbackbox.h	(working copy)
@@ -69,6 +69,7 @@
     void changeVolume(bool up_or_down);
     void changeSpeed(bool up_or_down);
     void toggleMute();
+    void toggleUpmix();
     void resetTimer();
     void hideVolume(){showVolume(false);}
     void showVolume(bool on_or_off);
Index: mythplugins/mythmusic/mythmusic/cddecoder.cpp
===================================================================
--- mythplugins/mythmusic/mythmusic/cddecoder.cpp	(revision 21650)
+++ mythplugins/mythmusic/mythmusic/cddecoder.cpp	(working copy)
@@ -14,6 +14,7 @@
 
 #include <mythtv/mythcontext.h>
 #include <mythtv/mythmediamonitor.h>
+#include <mythtv/libavcodec/avcodec.h>
 
 CdDecoder::CdDecoder(const QString &file, DecoderFactory *d, QIODevice *i,
                      AudioOutput *o) :
@@ -149,7 +150,8 @@
     if (output())
     {
         const AudioSettings settings(
-            16 /*bits*/, chan, freq, false /* AC3/DTS passthru */);
+            16 /*bits*/, chan, CODEC_ID_PCM_S16LE, freq,
+            false /* AC3/DTS passthru */);
         output()->Reconfigure(settings);
         output()->SetSourceBitrate(44100 * 2 * 16);
     }
