--- mythtv.orig/mythplugins/mythweather/mythweather/scripts/Makefile.inc	2010-12-05 17:48:10.000000000 -0500
+++ mythtv/mythplugins/mythweather/mythweather/scripts/Makefile.inc	2010-12-06 04:59:49.000000000 -0500
@@ -24,13 +24,13 @@
 
 install-scripts:
 	-[ "${SCRIPTS}" != "" ] && \
-	    ${INSTALL} -m 755 -d ${INSTDIR}/ && \
-	    for i in ${SCRIPTS} ; do ${INSTALL} -m 755 $$i ${INSTDIR}/ ; done
+	    ${INSTALL} -m 755 -d $(INSTALL_ROOT)${INSTDIR}/ && \
+	    for i in ${SCRIPTS} ; do ${INSTALL} -m 755 $$i $(INSTALL_ROOT)${INSTDIR}/ ; done
 
 install-data:
 	-[ "${DATA}" != "" ] && \
-	    ${INSTALL} -m 755 -d ${INSTDIR}/ && \
-	    for i in ${DATA} ; do ${INSTALL} -m 644 $$i ${INSTDIR}/ ; done
+	    ${INSTALL} -m 755 -d $(INSTALL_ROOT)${INSTDIR}/ && \
+	    for i in ${DATA} ; do ${INSTALL} -m 644 $$i $(INSTALL_ROOT)${INSTDIR}/ ; done
 
 uninstall-scripts:
 	-[ "${SCRIPTS}" != "" ] && \
