diff -Naur mythstream-0.18_qt4-old/parsers/apple.pl mythstream-0.18_qt4-new/parsers/apple.pl
--- mythstream-0.18_qt4-old/parsers/apple.pl	2009-08-15 11:57:37.000000000 -0700
+++ mythstream-0.18_qt4-new/parsers/apple.pl	2009-08-15 11:58:27.000000000 -0700
@@ -102,17 +102,15 @@
 	my $TRAILERS_URL_XQL   = 'VBoxView/MatrixView/VBoxView/HBoxView/TextView/GotoURL';
 	my $TRAILERS_TITLE_XQL = 'SetFontStyle/B/textNode()[0]';
 
-	trailers($maindoc, $TRAILERS_XQL, $TRAILERS_TITLE_XQL, '', $TRAILERS_URL_XQL);
+	getTrailers($maindoc, $TRAILERS_XQL, $TRAILERS_TITLE_XQL, '', $TRAILERS_URL_XQL);
 }
   elsif($command eq 'trailer')
 {
-   # en met bot geweld...
-	($url) = $response->content =~ /<key>previewURL<\/key><string>(.*?)<\/string>/;
-	($name) = $response->content =~ /<key>playlistName<\/key><string>(.*?)<\/string>/;
-
-   newItem(	name => $name,
-				url => $url,
-				handler => '');
+ 	my $TRAILER_XQL = '/Document/TrackList/plist/dict/array/dict';
+	my $TRAILER_URL_KEY = 'previewURL';
+	my $TRAILER_TITLE_KEY = 'songName';
+
+	getTrailer($maindoc, $TRAILER_XQL, $TRAILER_TITLE_KEY, $TRAILER_URL_KEY);
 }
 
 print $output_head->toString;
@@ -135,7 +133,7 @@
 	}
 }
 
-sub trailers
+sub getTrailers
 {
 	my $doc = shift;
 	my $root_xql = shift;
@@ -158,6 +156,27 @@
 	}
 }
 
+sub getTrailer
+{
+	my $doc = shift;
+	my $root_xql = shift;
+	my $title_key = shift;
+	my $url_key = shift;
+
+	foreach my $instance ($doc->xql($root_xql))
+	{
+		my $string = $instance->toString();
+		my ($title) = $string =~ /<key>$title_key<\/key><string>(.*?)<\/string>/;
+		my ($url)   = $string =~ /<key>$url_key<\/key><string>(.*?)<\/string>/;
+		if (($title) && ($url))
+		{
+			newItem(	name => $title,
+					url => $url,
+					handler => '');
+		}
+	}
+}
+
 sub trimText
 {
 	my $text = shift;
