diff -Naur mythstream-0.18_qt4-old/parsers/apple.pl mythstream-0.18_qt4-new/parsers/apple.pl
--- mythstream-0.18_qt4-old/parsers/apple.pl	2008-04-07 15:25:46.000000000 -0700
+++ mythstream-0.18_qt4-new/parsers/apple.pl	2009-07-07 16:43:02.000000000 -0700
@@ -108,7 +108,7 @@
 {
    # en met bot geweld...
 	($url) = $response->content =~ /<key>previewURL<\/key><string>(.*?)<\/string>/;
-	($name) = $response->content =~ /<key>podcastName<\/key><string>(.*?)<\/string>/;
+	($name) = $response->content =~ /<key>playlistName<\/key><string>(.*?)<\/string>/;
 
    newItem(	name => $name,
 				url => $url,
