diff -Naur mythstream-0.18_qt4-old/mythstream/mythstream.cpp mythstream-0.18_qt4-new/mythstream/mythstream.cpp
--- mythstream-0.18_qt4-old/mythstream/mythstream.cpp	2009-08-15 01:44:07.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/mythstream.cpp	2009-08-15 01:44:44.000000000 -0700
@@ -94,10 +94,10 @@
 {
 }
 
-void setPainterCol(QPainter *p, QColor col)
+void setPainterCol(QPainter &p, QColor col)
 {
-   p->setPen( col );
-   p->setBrush( QBrush(col, Qt::SolidPattern) );
+   p.setPen( col );
+   p.setBrush( QBrush(col, Qt::SolidPattern) );
 }
 
 void FFTBox::buildFFTBackground()
@@ -112,9 +112,11 @@
    fftPict = QPixmap(bckPict);
    QColor col = QColor(0x07, 0x99, 0);
 
-   QPainter *painter = new QPainter( &fftPict );
-   painter->setPen( col );
-   painter->setBrush( QBrush(col, Qt::SolidPattern) );
+   QPainter p;
+
+   p.begin( &fftPict );
+   p.setPen( col );
+   p.setBrush( QBrush(col, Qt::SolidPattern) );
 
    int barHeight = height();
    int divSpace = 2;
@@ -129,13 +131,13 @@
    for ( int i = 0; i < barCount; i++)
       for ( int j = 0; j < 20; j++)
       {
-      if (j == 0)  setPainterCol(painter, fftLow);
-      if (j == 5)  setPainterCol(painter, fftMid);
-      if (j == 10) setPainterCol(painter, fftHigh);
-      painter->drawRect( i*(barWidth + 2) + 1, barHeight - (j + 1) * (barDiv + divSpace) + divSpace, barWidth, barDiv);
+      if (j == 0)  setPainterCol(p, fftLow);
+      if (j == 5)  setPainterCol(p, fftMid);
+      if (j == 10) setPainterCol(p, fftHigh);
+      p.drawRect( i*(barWidth + 2) + 1, barHeight - (j + 1) * (barDiv + divSpace) + divSpace, barWidth, barDiv);
       }
 
-   delete(painter);
+   p.end();
 
    fftMask = QBitmap(fftPict.size(), true);
    drawPict = QPixmap(fftPict.size());
@@ -152,9 +154,11 @@
    fftMask.fill(Qt::color0);
    int barHeight;
 
-   QPainter *painter = new QPainter( &fftMask );
-   painter->setPen( col );
-   painter->setBrush( QBrush(col, Qt::SolidPattern) );
+   QPainter p;
+
+   p.begin( &fftMask );
+   p.setPen( col );
+   p.setBrush( QBrush(col, Qt::SolidPattern) );
 
    if (percentage)
    {
@@ -163,20 +167,25 @@
          barHeight = int ( float( percentage->values[i] ) * div );
          if ( barHeight > height() ) barHeight = height();
          if ( barHeight < 0 ) barHeight = 0;
-         painter->drawRect( i*(barWidth + 2) + 1, height() - barHeight, barWidth, height() );
+         p.drawRect( i*(barWidth + 2) + 1, height() - barHeight, barWidth, height() );
       }
    }
 
-   painter->setPen( Qt::color1 );
-   painter->setBrush( QBrush(Qt::color1, Qt::NoBrush) );
-   painter->drawRect( 0, height()-1, gaugeWidth, height() );
-   delete(painter);
+   p.setPen( Qt::color1 );
+   p.setBrush( QBrush(Qt::color1, Qt::NoBrush) );
+   p.drawRect( 0, height()-1, gaugeWidth, height() );
+   p.end();
 
    fftPict.setMask(fftMask);
 
-   bitBlt(&drawPict, QPoint(0,0), &bckPict, rect(), QPainter::CompositionMode_SourceOver);
-   bitBlt(&drawPict, QPoint(0,0), &fftPict, rect(), QPainter::CompositionMode_SourceOver);
-   bitBlt(this, QPoint(0,0), &drawPict, rect(), QPainter::CompositionMode_SourceOver);
+   p.begin( &drawPict );
+   p.drawPixmap(QPoint(0,0), bckPict, rect());
+   p.drawPixmap(QPoint(0,0), fftPict, rect());
+   p.end();
+
+   p.begin( this );
+   p.drawPixmap(QPoint(0,0), drawPict, rect());
+   p.end();
 }
 
 void FFTBox::setPercentage(Spectrum* percent)
@@ -801,7 +810,7 @@
         if (clearVideo)
         {
            clearOnly = true; // oops, gui is mess... (clear video window)
-           invalidateSection(bot);
+           invalidateSection(all);
            updateInvalidated();
            clearVideo = false;
         }
@@ -884,9 +893,15 @@
    if ( playerState.pollMe() )
    {
       if ( playerState.getDisplayStatus() == PlayerState::video )
+      {
         streamBrowser->showVideo();
+        invalidateSection(all);
+      }
+      else
+      {
+        invalidateSection(bot);
+      }
 
-      botSectInvalidated = true;
       updateInvalidated();
    }
 }
@@ -1285,7 +1300,7 @@
          {
             streamBrowser->hideVideo();      // hide video window
             clearOnly = true; // oops, gui is mess... (clear video window)
-            invalidateSection(bot);
+            invalidateSection(all);
             updateInvalidated();
          }
 
@@ -1511,9 +1526,13 @@
 // new screen update handling, partly implemented
 void MythStream::updateInvalidated()
 {
-     if (topSectInvalidated) update(m_TopRect);
-     if (midSectInvalidated) update(m_MidRect);
-     if (botSectInvalidated) update(m_BotRect);
+     QRect rect;
+     
+     if (topSectInvalidated) rect = rect | m_TopRect;
+     if (midSectInvalidated) rect = rect | m_MidRect;
+     if (botSectInvalidated) rect = rect | m_BotRect;
+
+     update(rect);
      
      topSectInvalidated = false;
      midSectInvalidated = false;
@@ -1546,6 +1565,7 @@
       case StreamBrowser::streamstatus:
          getStreamStatusValues();
          botSectInvalidated = true; // stream info fields
+         midSectInvalidated = true;
          topSectInvalidated = true; // status field
       break;
       case StreamBrowser::streamplay:
@@ -1816,11 +1836,14 @@
 {
 
    // unconditional update
-   QPixmap pix(m_TopRect.size());
-   pix.fill(this, m_TopRect.topLeft());
-   QPainter p(&pix);
 
    LayerSet* container = m_Theme->GetSet("status_panel");
+   QRect rect = container->GetAreaRect();
+
+   QPainter p;
+
+   QPixmap pix( rect.size() );
+   pix.fill(this, rect.topLeft());
 
    if (container)
    {
@@ -1837,8 +1860,9 @@
 
    p.end();
 
-   bitBlt(this, m_TopRect.left(), m_TopRect.top(),
-         &pix, 0, 0, -1, -1, QPainter::CompositionMode_SourceOver);
+   p.begin(this);
+   p.drawPixmap(rect.left(), rect.top(), pix);
+   p.end();
 
 }
 
@@ -1851,10 +1875,11 @@
    LayerSet* container = m_Theme->GetSet("dyn_panel");
    QRect rect = container->GetAreaRect();
 
+   QPainter p;
+
    QPixmap pix( rect.size() );
    pix.fill(this, rect.topLeft() );
-   QPainter p(&pix);
-
+   p.begin(&pix);
 
    if (container)
    {
@@ -1864,8 +1889,9 @@
 
    p.end();
 
-   bitBlt(this, rect.left(), rect.top(),
-         &pix, 0, 0, -1, -1, QPainter::CompositionMode_SourceOver);
+   p.begin(this);
+   p.drawPixmap(rect.left(), rect.top(), pix);
+   p.end();
 
 }
 
@@ -1902,9 +1928,11 @@
 
    QRect rect = container->GetAreaRect();
 
+   QPainter p;
+
    QPixmap pix( rect.size() );
    pix.fill(this, rect.topLeft());
-   QPainter p(&pix);
+   p.begin(&pix);
 
    if (container)
    {
@@ -1925,8 +1953,9 @@
 
    p.end();
 
-   bitBlt(this, rect.left(), rect.top(),
-         &pix, 0, 0, -1, -1, QPainter::CompositionMode_SourceOver);
+   p.begin(this);
+   p.drawPixmap(rect.left(), rect.top(), pix);
+   p.end();
 
 }
 
