diff -Naur mythstream-0.18_qt4-old/mythstream/mythstream.cpp mythstream-0.18_qt4-new/mythstream/mythstream.cpp
--- mythstream-0.18_qt4-old/mythstream/mythstream.cpp	2009-08-15 01:41:53.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/mythstream.cpp	2009-08-15 01:44:07.000000000 -0700
@@ -327,6 +327,8 @@
 
    clearVideo = false; // was video screen drawn?
 
+   clearOnly = false;  // should the next paint be clear only.
+
    setNoErase();
    updateBackground();
 
@@ -798,7 +800,9 @@
         
         if (clearVideo)
         {
-           updateBotView(true); // oops, gui is mess... (clear video window)
+           clearOnly = true; // oops, gui is mess... (clear video window)
+           invalidateSection(bot);
+           updateInvalidated();
            clearVideo = false;
         }
       }
@@ -1280,7 +1284,9 @@
          else
          {
             streamBrowser->hideVideo();      // hide video window
-            updateBotView(true); // oops, gui is mess... (clear video window)
+            clearOnly = true; // oops, gui is mess... (clear video window)
+            invalidateSection(bot);
+            updateInvalidated();
          }
 
       }
@@ -1573,15 +1579,30 @@
 {
    QRect r = e->rect();
 
+   // the video panel is not clean so clean it
+   if (clearOnly)
+   {
+      updateBotView();
+      clearOnly = false;
+      // if we are playing a video, then there is nothing left to update,
+      // although I do not know why we would clearing the video panel were a video playing
+      if ( playerState.getDisplayStatus() == PlayerState::video )
+      {
+         return;
+      }
+   }
+
    if (r.intersects(m_TopRect))
+   {
       updateTopView();
+   }
    if (r.intersects(m_MidRect))
    {
       updateMidView();
    }
    if (r.intersects(m_BotRect))
    {
-      updateBotView(false);
+      updateBotView();
    }
 }
 
@@ -1849,7 +1870,7 @@
 }
 
 
-void MythStream::updateBotView(bool clearOnly)
+void MythStream::updateBotView()
 {
 
    LayerSet* container = 0;
diff -Naur mythstream-0.18_qt4-old/mythstream/mythstream.h mythstream-0.18_qt4-new/mythstream/mythstream.h
--- mythstream-0.18_qt4-old/mythstream/mythstream.h	2009-08-15 01:23:04.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/mythstream.h	2009-08-15 01:44:07.000000000 -0700
@@ -197,7 +197,7 @@
     
     void updateTopView();
     void updateMidView();
-    void updateBotView(bool clearOnly);
+    void updateBotView();
 
     void keyPressEvent( QKeyEvent *k );
 
@@ -232,6 +232,7 @@
     unsigned int m_FolderDisplaySize;
     bool clearVideo;
     bool harvesting;
+    bool clearOnly;
 
 
     FieldOwner messageOwner;
