diff -u -r mythtv.orig/configure mythtv/configure
--- mythtv.orig/configure	2010-08-28 10:46:53.000000000 -0400
+++ mythtv/configure	2010-08-28 10:48:43.000000000 -0400
@@ -4749,9 +4749,9 @@
 cat >> external/FFmpeg/config.mak <<EOF
 LIBDIR=\$(INSTALL_ROOT)$libdir
 SHLIBDIR=\$(INSTALL_ROOT)$shlibdir
-INCDIR=\$(INSTALL_ROOT)$incdir
-BINDIR=\$(INSTALL_ROOT)$bindir
-DATADIR=\$(INSTALL_ROOT)$datadir
+INCDIR=$incdir
+BINDIR=$bindir
+DATADIR=$datadir
 MANDIR=\$(INSTALL_ROOT)$mandir
 endif # FFMPEG_CONFIG_MAK
 EOF
