diff --git a/mythtv-status/bin/mythtv-update-motd b/mythtv-status/bin/mythtv-update-motd
index d58aa41..654bb13 100755
--- a/mythtv-status/bin/mythtv-update-motd
+++ b/mythtv-status/bin/mythtv-update-motd
@@ -1,16 +1,16 @@
 #!/bin/bash

-. /etc/default/mythtv-status
+args=("$@")

 if [ x$RUN != xyes ]
 then
   exit
 fi

-[ ! -f /var/run/motd.orig ] && cp /var/run/motd /var/run/motd.orig
+[ ! -f /etc/motd.orig ] && cp /etc/motd /etc/motd.orig

-cp /var/run/motd.orig /var/run/motd.new
-if mythtv-status $ARGS -h $HOST >> /var/run/motd.new
-then
-  mv /var/run/motd.new /var/run/motd
+cp /etc/motd.orig /etc/motd.new
+if mythtv-status $@ -h $HOST >> /etc/motd.new then
+  mv /etc/motd.new /etc/motd
 fi

