diff -u -r a/bin/mythtv-update-motd b/bin/mythtv-update-motd
--- a/bin/mythtv-update-motd	2010-09-17 15:17:21.000000000 -0400
+++ b/bin/mythtv-update-motd	2010-09-17 15:40:50.000000000 -0400
@@ -7,11 +7,11 @@
   exit
 fi
 
-[ ! -f /var/run/motd.orig ] && cp /var/run/motd /var/run/motd.orig
+[ ! -f /etc/motd.orig ] && cp /etc/motd /etc/motd.orig
 
-cp /var/run/motd.orig /var/run/motd.new
+cp /etc/motd.orig /etc/motd.new
 ret=0
-mythtv-status $ARGS -h $HOST >> /var/run/motd.new 2> /dev/null || ret=$?
+mythtv-status $ARGS -h $HOST >> /etc/motd.new 2> /dev/null || ret=$?
 if [ $ret -eq 0 -o $ret -eq 1 ]; then
-  [ -f /var/run/motd.new ] && mv /var/run/motd.new /var/run/motd
+  [ -f /etc/motd.new ] && mv /etc/motd.new /etc/motd
 fi
