diff -Naur mythstream-0.18_qt4-old/libs/configelements.cpp mythstream-0.18_qt4-new/libs/configelements.cpp
--- mythstream-0.18_qt4-old/libs/configelements.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/configelements.cpp	2009-08-11 16:20:39.000000000 -0700
@@ -110,7 +110,7 @@
       lab->setMaximumHeight(maxHeight);
       tmp = new MyLineEdit(this, "edit" + QString::number(i));
 #ifdef MYTHTV      
-      tmp->setPopupPosition(VK_POSTOPDIALOG);
+      tmp->setPopupPosition(VKQT_POSTOPDIALOG);
 #endif
     }
     
diff -Naur mythstream-0.18_qt4-old/libs/configelements.h mythstream-0.18_qt4-new/libs/configelements.h
--- mythstream-0.18_qt4-old/libs/configelements.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/configelements.h	2009-08-11 16:20:39.000000000 -0700
@@ -30,7 +30,7 @@
 
 #ifdef MYTHTV
 #include <mythtv/mythwidgets.h>
-#include <mythtv/virtualkeyboard.h>
+#include <mythtv/virtualkeyboard_qt.h>
 #else
 #include <qlineedit.h>
 #endif
diff -Naur mythstream-0.18_qt4-old/libs/recorder.cpp mythstream-0.18_qt4-new/libs/recorder.cpp
--- mythstream-0.18_qt4-old/libs/recorder.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/recorder.cpp	2009-08-11 16:20:39.000000000 -0700
@@ -112,7 +112,7 @@
    if (recList.count() != 0) // no storage event should occur during recording
      cerr << "TARGET warning: storage manipulation during recording" << endl;
 
-   if (eventType == Storage::loaded)
+   if (eventType == MStorage::loaded)
    {
       if (recList.count() != 0) // no storage event should occur during recording
         stopAllRecordings();
diff -Naur mythstream-0.18_qt4-old/libs/storage.cpp mythstream-0.18_qt4-new/libs/storage.cpp
--- mythstream-0.18_qt4-old/libs/storage.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/storage.cpp	2009-08-11 16:20:39.000000000 -0700
@@ -37,9 +37,9 @@
 using namespace std;
 
 
-//----------------------- Storage ---------------------------
+//----------------------- MStorage ---------------------------
 
-Storage::Storage(QString table, QString name,
+MStorage::MStorage(QString table, QString name,
                  int key_pri, int key_mul1, int key_mul2) : QObject()
 {
     currentStorage = 0;
@@ -53,7 +53,7 @@
     this->key_mul2 = key_mul2;
 }
 
-Storage::~Storage()
+MStorage::~MStorage()
 {
   if (currentStorage)
     delete currentStorage;
@@ -62,7 +62,7 @@
 }
 
 
-bool Storage::storageReadOnly()
+bool MStorage::storageReadOnly()
 {
   if (currentStorage)
     return currentStorage->storageReadOnly();
@@ -70,7 +70,7 @@
     return false;
 }
 
-QString Storage::getStorageDescription()
+QString MStorage::getStorageDescription()
 {
   if (currentStorage)
     return currentStorage->getStorageDescription();
@@ -78,7 +78,7 @@
     return "TARGET storage warning: no storage selected";
 }
 
-Storage::AccessType Storage::getAccessType()
+MStorage::AccessType MStorage::getAccessType()
 {
   if (currentStorage)
     return (AccessType)currentStorage->getAccessType();
@@ -86,7 +86,7 @@
     return none;
 }
 
-QString Storage::getStorageName()
+QString MStorage::getStorageName()
 {
   if (currentStorage)
     return currentStorage->getStorageName();
@@ -94,7 +94,7 @@
     return "warning: no storage selected";
 }
 
-bool Storage::getSynchronized()
+bool MStorage::getSynchronized()
 {
   if (currentStorage)
     return currentStorage->getSynchronized();
@@ -102,7 +102,7 @@
     return false;
 }
 
-QString Storage::getLastError()
+QString MStorage::getLastError()
 {
   if (currentStorage)
     return currentStorage->getLastError();
@@ -110,13 +110,13 @@
     return "";
 }
 
-void Storage::resetRecordList()
+void MStorage::resetRecordList()
 {
   if (currentStorage)
     currentStorage->resetRecordList();
 }
 
-bool Storage::getNextRecord(ValueList& values)
+bool MStorage::getNextRecord(ValueList& values)
 {
   if (currentStorage)
     return currentStorage->getNextRecord(currentStorage->getListHandler(), values);
@@ -124,7 +124,7 @@
     return false;
 }
 
-void Storage::linkNewStorage(GenStorage *storage)
+void MStorage::linkNewStorage(GenStorage *storage)
 {
     currentStorage = storage;
 
@@ -158,7 +158,7 @@
 
 }
 
-bool Storage::selectDbStorage(int ident, QString name, QString dbDriver,
+bool MStorage::selectDbStorage(int ident, QString name, QString dbDriver,
                               QString hostName, unsigned int port, QString dbName, QString table,
                               QString login, QString password)
 {
@@ -175,7 +175,7 @@
 }
 
 #ifdef MYTHTV
-bool Storage::selectDefaultDb(int ident)
+bool MStorage::selectDefaultDb(int ident)
 {
     if (oldStorage) delete oldStorage;
     oldStorage = currentStorage;
@@ -189,7 +189,7 @@
 }
 #endif
 
-bool Storage::selectFileStorage(int ident, QString name, QString fileName)
+bool MStorage::selectFileStorage(int ident, QString name, QString fileName)
 {
     if (oldStorage) delete oldStorage;
     oldStorage = currentStorage;
@@ -202,7 +202,7 @@
     return fileStorage->openFileStorage(ident, fileName);
 }
 
-void Storage::selectWebStorage(int ident, QString name, QString url, QString login, QString password)
+void MStorage::selectWebStorage(int ident, QString name, QString url, QString login, QString password)
 {
 
     if (oldStorage) delete oldStorage;
@@ -217,7 +217,7 @@
 }
 
 
-bool Storage::loadList(int ident, QString& error)
+bool MStorage::loadList(int ident, QString& error)
 {
     if (currentStorage)
     {
@@ -238,7 +238,7 @@
 
 // - copy from resource1 to resource2
 // - update file resource after removal
-bool Storage::storeList(int ident, QString& error)
+bool MStorage::storeList(int ident, QString& error)
 {
    if (!currentStorage)
    {
@@ -255,7 +255,7 @@
    return currentStorage->storeList(ident, oldStorage->getListHandler(), error);
 }
 
-bool Storage::insertRecord(int ident, ValueList& values, QString& error)
+bool MStorage::insertRecord(int ident, ValueList& values, QString& error)
 {
    if (currentStorage)
      return currentStorage->insertRecord(ident, values, error);
@@ -266,7 +266,7 @@
    }
 }
 
-bool Storage::updateRecord(int ident, ValueList& oldValues, ValueList& values, QString& error)
+bool MStorage::updateRecord(int ident, ValueList& oldValues, ValueList& values, QString& error)
 {
 
    if (currentStorage)
@@ -278,7 +278,7 @@
    }
 }
 
-bool Storage::removeRecord(int ident, ValueList values, QString& error)
+bool MStorage::removeRecord(int ident, ValueList values, QString& error)
 {
    if (currentStorage)
      return currentStorage->removeRecord(ident, values, error);
diff -Naur mythstream-0.18_qt4-old/libs/storage.h mythstream-0.18_qt4-new/libs/storage.h
--- mythstream-0.18_qt4-old/libs/storage.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/storage.h	2009-08-11 16:20:39.000000000 -0700
@@ -37,7 +37,9 @@
 /*
   Absolutely NOT thread safe
 
-  Storage constraints:
+  Storage has been renamed to MStorage: it conflicted with new mythtv class Storage
+
+  MStorage constraints:
     1) url is unique
     2) (folder, name) is unique
 
@@ -53,7 +55,7 @@
       -> In case of file storage: selectFileStorage + removeRecord + storeList compacts file
 */
 
-class Storage : public QObject
+class MStorage : public QObject
 {
     Q_OBJECT
 
@@ -74,9 +76,9 @@
       saved
     };
 
-    Storage(QString table = "", QString name = "",
+    MStorage(QString table = "", QString name = "",
             int key_pri = 2, int key_mul1 = 0, int key_mul2 = 1);
-    ~Storage();
+    ~MStorage();
 
     // select storage resource (without reading records)
 #ifdef MYTHTV
diff -Naur mythstream-0.18_qt4-old/libs/streamstorage.cpp mythstream-0.18_qt4-new/libs/streamstorage.cpp
--- mythstream-0.18_qt4-old/libs/streamstorage.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamstorage.cpp	2009-08-11 16:20:39.000000000 -0700
@@ -326,7 +326,7 @@
     return (values.count() >= r_count_file && values[r_name] == storageName);
 }
 
-ReposStorage::ReposStorage() : Storage("", "", 2, 2, 2)
+ReposStorage::ReposStorage() : MStorage("", "", 2, 2, 2)
 {
     QString error;
     QString source;
diff -Naur mythstream-0.18_qt4-old/libs/streamstorage.h mythstream-0.18_qt4-new/libs/streamstorage.h
--- mythstream-0.18_qt4-old/libs/streamstorage.h	2008-04-07 04:17:57.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/streamstorage.h	2009-08-11 16:20:39.000000000 -0700
@@ -36,15 +36,15 @@
 
 // Open reposstorage first: it creates default files on first run
 
-class StreamStorage : public Storage
+class StreamStorage : public MStorage
 {
   public:
-    StreamStorage(QString table = "", QString name = "") : Storage(table, name) {}
+    StreamStorage(QString table = "", QString name = "") : MStorage(table, name) {}
 
     bool selectStorage(int ident, ValueList& values);
 };
 
-class ReposStorage : public Storage
+class ReposStorage : public MStorage
 {
   public:
     ReposStorage();
diff -Naur mythstream-0.18_qt4-old/libs/x11win.cpp mythstream-0.18_qt4-new/libs/x11win.cpp
--- mythstream-0.18_qt4-old/libs/x11win.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/libs/x11win.cpp	2009-08-11 18:17:05.000000000 -0700
@@ -1,26 +1,28 @@
 
 #include <iostream>
 #include <X11/Xlib.h>
+#ifdef CursorShape
+#undef CursorShape // defined as a constant in X11/X.h and used as an enum in QtGui/qcursor.h.
+#endif
+#include <QX11Info>
 
 
 using namespace std;
 
 //extern void handleExposeEvent();
 
-Display* qt_xdisplay (void);
-
 void myX11MapWindow (uint id)
 {
   //cout << "mapping" << endl;
-  XMapWindow (qt_xdisplay(), id);
-  XFlush (qt_xdisplay());
+  XMapWindow (QX11Info::display(), id);
+  XFlush (QX11Info::display());
 }
 
 void myX11UnmapWindow (uint id)
 {
  // cout << "unmapping" << endl;
-  XUnmapWindow (qt_xdisplay(), id);
-  XFlush (qt_xdisplay());
+  XUnmapWindow (QX11Info::display(), id);
+  XFlush (QX11Info::display());
 }
 
 
@@ -28,7 +30,7 @@
 {
    uint res;
    
-   res = XGrabKeyboard(qt_xdisplay(), id, true, GrabModeAsync, GrabModeAsync, CurrentTime);
+   res = XGrabKeyboard(QX11Info::display(), id, true, GrabModeAsync, GrabModeAsync, CurrentTime);
 
    switch (res)
    {
@@ -43,7 +45,7 @@
 
 void myX11UngrabKeyboard()
 {
-   XUngrabKeyboard(qt_xdisplay(), CurrentTime);
+   XUngrabKeyboard(QX11Info::display(), CurrentTime);
 }
 
 /*
diff -Naur mythstream-0.18_qt4-old/mythstream/main.cpp mythstream-0.18_qt4-new/mythstream/main.cpp
--- mythstream-0.18_qt4-old/mythstream/main.cpp	2008-04-07 14:14:56.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream/main.cpp	2009-08-11 16:20:39.000000000 -0700
@@ -32,11 +32,12 @@
 #include "../libs/streamconfig.h"
 #include "../libs/storageconfig.h"
 
-#include <mythtv/mythcontext.h>
 #include <mythtv/mythdialogs.h>
 #include <mythtv/mythmedia.h>
 #include <mythtv/mythplugin.h>
+#include <mythtv/libmythdb/mythversion.h>
 #include <mythtv/libmythui/myththemedmenu.h>
+#include <mythtv/libmythui/mythuihelper.h>
 
 using namespace std;
 
@@ -147,10 +148,10 @@
 
 void runStream(void)
 {
-    gContext->addCurrentLocation("mythstream");
+    GetMythUI()->AddCurrentLocation("mythstream");
     MythStream stream(gContext->GetMainWindow(), "stream");
     stream.exec();
-    gContext->removeCurrentLocation( );
+    GetMythUI()->RemoveCurrentLocation( );
 }
 
 int mythplugin_run(void)
@@ -162,7 +163,7 @@
 
 int mythplugin_config(void)
 {
-    QString themedir = gContext->GetThemeDir( );
+    QString themedir = GetMythUI()->GetThemeDir( );
     runMenu( themedir, "streamconfigmenu.xml");
     return 0;
 }
diff -Naur mythstream-0.18_qt4-old/mythstream.pro mythstream-0.18_qt4-new/mythstream.pro
--- mythstream-0.18_qt4-old/mythstream.pro	2008-04-07 05:28:59.000000000 -0700
+++ mythstream-0.18_qt4-new/mythstream.pro	2009-08-11 18:05:27.000000000 -0700
@@ -3,4 +3,3 @@
 # Directories
 SUBDIRS = mythstream
 #SUBDIRS = mythstream alt_parsers
-
diff -Naur mythstream-0.18_qt4-old/settings.pro mythstream-0.18_qt4-new/settings.pro
--- mythstream-0.18_qt4-old/settings.pro	2008-04-07 15:40:18.000000000 -0700
+++ mythstream-0.18_qt4-new/settings.pro	2009-08-11 16:20:39.000000000 -0700
@@ -3,15 +3,26 @@
 
 VERSION=0.18_2
 
-# Install in /usr/share/... or /usr/local/share/... ?
-#PREFIX = /usr/local
-PREFIX = /usr
+# Where binaries, includes and runtime assets are installed by 'make install'
+isEmpty( PREFIX ) {
+    # Install in /usr/share/... or /usr/local/share/... ?
+    #PREFIX = /usr/local
+    PREFIX = /usr
+}
+# Where the binaries actually locate the assets/filters/plugins at runtime
+isEmpty( RUNPREFIX ) {
+    RUNPREFIX = $$PREFIX
+}
 
 INCLUDEPATH += $${PREFIX}/include
+INCLUDEPATH += $${PREFIX}/include/mythtv
+INCLUDEPATH += $${PREFIX}/include/mythtv/libmythdb
+INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += $${PREFIX}/include/mythtv/libmyth
 #INCLUDEPATH *= /usr/local/include
 
 DEFINES += _GNU_SOURCE
-DEFINES += PREFIX=\\\"$${PREFIX}\\\"
+DEFINES += PREFIX=\\\"$${RUNPREFIX}\\\"
 DEFINES += VERSION=\\\"$${VERSION}\\\"
 
 QMAKE_LFLAGS -= -Wl,--no-undefined
